internal/internal.go

```diff
@@ -59,18 +61,59 @@ type StorageConfig struct {
 	Bucket string
 }
 
-// MockTokenSource is a TokenSource implementation that can be used for testing.
-type MockTokenSource struct {
-	AccessToken string
-}
-
 // MessagingConfig represents the configuration of Firebase Cloud Messaging service.
 type MessagingConfig struct {
 	Opts      []option.ClientOption
 	ProjectID string
 	Version   string
 }
 
+// FirebaseError is an error type containing an error code string.
+type FirebaseError struct {
+	Code   string
+	String string
+}
+
+func (fe *FirebaseError) Error() string {
+	return fe.String
+}
+
+// HasErrorCode checks if the given error contain a specific error code.
+func HasErrorCode(err error, code string) bool {
+	fe, ok := err.(*FirebaseError)
+	return ok && fe.Code == code
+}
+
+// Error creates a new FirebaseError from the specified error code and message.
+func Error(code string, msg string) *FirebaseError {
+	return &FirebaseError{
+		Code:   code,
+		String: msg,
+	}
+}
+
+// Errorf creates a new FirebaseError from the specified error code and message.
+func Errorf(code string, msg string, args ...interface{}) *FirebaseError {
+	return Error(code, fmt.Sprintf(msg, args...))
+}
+
+// Code extracts the error code from the given error.
+//
+// If the error does not contain an error code, this function returns an empty string. Use this
+// internal function for unit tests in packages that cannot directly import the root firebase
+// package.
+func Code(err error) string {
```

Comment:
It looks like this isn't actually used anywhere?

Reply:
Removed